### DISCUSSION TITLE
SafeMath usage in solidity

### DISCUSSION THREAD

**Discussion by geniusdev329** [community/community#51069] [Codespaces] [answered]
### Select Topic Area

Product Feedback

### Body

I used safemath library in smart contract development but some developers said it is not needed in high versions.
Do you correct the version that it is not needed in?

**Comment by cargood** [comment, marked as answer]
The safemath library is very important in smart contract. 
You can use to prevent the errors of underflow, overflow.
Actually, you don't need it in 0.8.0+ versions.

Also, you can use "unchecked" and "checked" for some cases.
